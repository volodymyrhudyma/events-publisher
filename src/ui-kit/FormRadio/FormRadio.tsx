import React from 'react';
import { IFormControlProps, ISelectOption } from 'src/types';
import FormControl from 'src/ui-kit/FormControl';
import Radio from 'src/ui-kit/Radio';
import { ChoiceContainer } from './styled/FormRadio.styled';

interface IFormRadioProps extends IFormControlProps {
	options: ISelectOption[];
	value: string;
	onChange: (value: string) => void;
}

const FormRadio = ({ options, value, onChange, ...rest }: IFormRadioProps) => (
	<FormControl {...rest}>
		<ChoiceContainer>
			{options.map((option) => (
				<Radio
					key={option.value}
					value={option.value}
					label={option.label}
					checked={option.value === value}
					onChange={onChange}
				/>
			))}
		</ChoiceContainer>
	</FormControl>
);

FormRadio.defaultProps = {
	required: false
};

export default FormRadio;
