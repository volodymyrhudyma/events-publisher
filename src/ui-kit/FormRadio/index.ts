import Component from './FormRadio';

export default Component;
