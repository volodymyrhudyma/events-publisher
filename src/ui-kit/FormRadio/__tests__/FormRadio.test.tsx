import React from 'react';
import { shallow } from 'enzyme';
import FormRadio from '../FormRadio';

describe('FormRadio component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <FormRadio
        label="Label"
        options={[
          { value: 'free', label: 'Free event' },
          { value: 'paid', label: 'Paid event' },
        ]}
        value=""
        onChange={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
