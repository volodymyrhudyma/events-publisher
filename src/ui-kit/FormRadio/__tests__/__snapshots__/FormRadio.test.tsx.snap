// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRadio component should render correctly 1`] = `
<FormControl
  label="Label"
  required={false}
>
  <styled.div>
    <Radio
      checked={false}
      key="free"
      label="Free event"
      onChange={[MockFunction]}
      value="free"
    />
    <Radio
      checked={false}
      key="paid"
      label="Paid event"
      onChange={[MockFunction]}
      value="paid"
    />
  </styled.div>
</FormControl>
`;
